components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/ServiceDiscovererKubernetes.java

```diff
@@ -0,0 +1,396 @@
+                    }
+                    // A Special case (can be any of the service types above)
+                    addExternalIPEndpoint(serviceName, serviceSpec.getExternalIPs(), port, protocol,
+                            namespace, labels, endpointList);
+                } else if (log.isDebugEnabled()) {
+                    log.debug("Service:{} Namespace:{} Port:{}/{}  Application level protocol not defined.",
+                            serviceName, namespace, servicePort.getPort(), protocol);
+                }
+            }
+        }
+    }
+
+
+    private void addExternalNameEndpoint(String serviceName, String externalName,
+                                         String namespace, String labels, List<Endpoint> endpointList)
+            throws MalformedURLException {
+        URL url = new URL("http://" + externalName);
+        endpointList.add(constructEndpoint(serviceName, namespace, "http", EXTERNAL_NAME, url, labels));
+    }
+
+    private void addClusterIPEndpoint(String serviceName, String clusterIP, int port, String protocol,
+                                      String namespace, String labels, List<Endpoint> endpointList)
+            throws MalformedURLException {
+        URL url = new URL(protocol, clusterIP, port, "");
+        endpointList.add(constructEndpoint(serviceName, namespace, protocol, CLUSTER_IP, url, labels));
+    }
+
+    private void addNodePortEndpoint(String serviceName, int nodePort, String protocol,
+                                     String namespace, String labels, List<Endpoint> endpointList)
+            throws MalformedURLException {
+        String nodeIP = client.getMasterUrl().getHost();
+        URL url = new URL(protocol, nodeIP, nodePort, "");
+        endpointList.add(constructEndpoint(serviceName, namespace, protocol, NODE_PORT, url, labels));
+    }
+
+    private void addLoadBalancerEndpoint(String serviceName, Service service, int port, String protocol,
+                                         String namespace, String labels, List<Endpoint> endpointList)
+            throws MalformedURLException {
+        List<LoadBalancerIngress> loadBalancerIngresses = service.getStatus().getLoadBalancer().getIngress();
+        if (!loadBalancerIngresses.isEmpty()) {
+            for (LoadBalancerIngress loadBalancerIngress : loadBalancerIngresses) {
+                String hostname = loadBalancerIngress.getHostname();
+                String host = (hostname == null || "".equals(hostname)) ? loadBalancerIngress.getIp() : hostname;
+                URL url = new URL(protocol, host, port, "");
+                endpointList.add(constructEndpoint(serviceName, namespace, protocol, LOAD_BALANCER, url, labels));
+            }
+        } else {
+            log.debug("Service:{}  Namespace:{}  Port:{}/{} has no loadBalancer ingresses available.",
+                    serviceName, namespace, port, protocol);
+        }
+    }
+
```

Comment:
There are multiple places here to add method comments.  Could you please check and add

Reply:
Sorry, I thought since they were private methods and the method name was long already, method comments are not necessary. I'll check all the classes added and add method comments.